app/api.py

```diff
@@ -170,16 +171,27 @@ def post(self, ext_mongo):
             return auth
         current_time = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
         indicators = args.get('indicators', list())
-        print indicators
         indicators = list(set(indicators))
+        tags = args.get('tags', list())
         for item in indicators:
-            #  Expecting MD5 indicators, so hash everything else.
-            if len(item) != 32:
-                item = extract_fqdn(item)  # Just in case
-                item = hashlib.md5(item).hexdigest()
-            obj = {'indicator': item, 'creator': auth['user']['email'],
-                   'datetime': current_time}
-            ext_mongo.db.indicators.insert_one(obj)
+            # Check if the indicator is already hashed.
+            if re.search(r"^([a-fA-F\d]{32})$", item):
+                orig = ""
+                hashed = item
+            # Otherwise it's an IOC in clear, we hash it and store the original.
+            else:
+                orig = extract_fqdn(item)
+                # We hash it.
+                hashed = hashlib.md5(orig).hexdigest()
+
+            # Look-up hashed IOC.
+            record = ext_mongo.db.indicators.find_one({'indicator': hashed})
```

Comment:
We can technically get around this by throwing a unique index on the indicator itself to avoid duplicate entry. Doesn't hurt to have the checks in code.